src/v/archival/ntp_archiver_service.cc

```diff
@@ -313,6 +315,21 @@ ss::future<cloud_storage::download_result> ntp_archiver::sync_manifest() {
     __builtin_unreachable();
 }
 
+void ntp_archiver::update_probe() {
+    const auto& man = manifest();
+
+    _probe.segments_in_manifest(man.size());
+
+    const auto first_addresable = man.first_addresable_segment();
+    const auto truncated_seg_count = first_addresable == man.end()
```

Comment:
This isn't too painful as it's once per update rather than each poll, but still a bit of a heavyweight op for stats.  Perhaps a motivation for refactoring the truncation code to move segments out of the main map and into _replaced sooner, in future.

Reply:
I don't think it's too bad as it is logarithmic in the size of the manifest, but I agree that it would be nice to have this in `replaced` instead.